third_party/openocd/reset_on_dmi_op_error.patch

```diff
@@ -0,0 +1,60 @@
+diff --git a/src/target/riscv/riscv-013.c b/src/target/riscv/riscv-013.c
+index 4e6c8dc36..94d178d06 100644
+--- a/src/target/riscv/riscv-013.c
++++ b/src/target/riscv/riscv-013.c
+@@ -538,6 +538,33 @@ static dmi_status_t dmi_scan(struct target *target, uint32_t *address_in,
+ 	return buf_get_u32(in, DTM_DMI_OP_OFFSET, DTM_DMI_OP_LENGTH);
+ }
+ 
++/**
++ * The OpenTitan rv_dm can get stuck in certain circumstances. This can easily
++ * be detected because tdo is stuck at one so reading DTMCS returns all ones,
++ * a clearly invalid value. Also the rv_dm never sends back an operation status
++ * other than 0 (success) or 3 (busy). When this happens, continue reading until it gets
++ * unstuck and performs a hard dmi reset.
++ */
++static int ott_recover(struct target *target)
```

Comment:
What does `ott` abbreviate in this context? I've seen it as abbreviation for opentitantool, but this has nothing to do with opentitantool, right?

(Also in the log messages below as `OTT`)

Reply:
Yes I was thinking opentitantool because that's all the infrastructure we are using for testing but I can change the name if you want.

Reply:
I see, thanks for changing it in the function name. I think it should also be changed in the log messages